src/hal/src/pso/graphics.rs

```diff
@@ -6,44 +6,55 @@ use super::input_assembler::{AttributeDesc, InputAssemblerDesc, VertexBufferDesc
 use super::output_merger::{ColorBlendDesc, DepthStencilDesc};
 
 /// A complete set of shaders to build a graphics pipeline.
+///
+/// All except the vertex shader are optional.
+/// DOC TODO: What happens if, say, a fragment shader is not defined?
```

Comment:
> If a fragment shader is omitted, the results of fragment processing are undefined. Specifically, any fragment color outputs are considered to have undefined values, and the fragment depth is considered to be unmodified. This can be useful for depth-only rendering.